independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ArcContainerImpl.java

```diff
@@ -149,6 +153,10 @@ public ArcContainerImpl(CurrentContextFactory currentContextFactory) {
             qualifierNonbindingMembers.putAll(components.getQualifierNonbindingMembers());
         }
 
+        List<InjectableContext> requestContexts = contexts.get(RequestScoped.class);
+        this.requestContextSingleton = requestContexts.size() == 1 ? Collections.singletonList(requestContexts.get(0)) : null;
```

Comment:
Why would you store `null` when you could store the list?

Reply:
Sorry, that was an old version...